app/src/main/java/com/android/shelfLife/ui/recipes/Recipes.kt

```diff
@@ -0,0 +1,373 @@
+) {
+  // Collect the recipes StateFlow as a composable state
+  val recipeList by listRecipesViewModel.recipes.collectAsState()
+
+  // State for the search query
+  var query by remember { mutableStateOf("") }
+
+  val selectedHousehold by householdViewModel.selectedHousehold.collectAsState()
+  val userHouseholds = householdViewModel.households.collectAsState().value
+
+  var showDialog by remember { mutableStateOf(false) }
+  var showEdit by remember { mutableStateOf(false) }
+
+  val drawerState = rememberDrawerState(DrawerValue.Closed)
+  val scope = rememberCoroutineScope()
+
+  val filters =
+      listOf(
+          "No missing ingredients",
+          "Vegan",
+          "Vegetarian",
+          "Easy",
+          "Spicy",
+          "Dessert",
+          "Appetizer",
+          "Snack",
+          "Breakfast")
+
+  AddHouseHoldPopUp(
+      showDialog = showDialog,
+      onDismiss = { showDialog = false },
+      householdViewModel = householdViewModel,
+  )
+
+  EditHouseHoldPopUp(
+      showDialog = showEdit,
+      onDismiss = { showEdit = false },
+      householdViewModel = householdViewModel)
+
+  ModalNavigationDrawer(
+      drawerState = drawerState,
+      drawerContent = {
+        ModalDrawerSheet {
+          Text(
+              "Household selection",
+              modifier =
+                  Modifier.padding(vertical = 18.dp, horizontal = 16.dp)
+                      .padding(horizontal = 12.dp),
+              style = MaterialTheme.typography.labelMedium)
+          userHouseholds.forEach { household ->
+            selectedHousehold?.let {
+              HouseHoldElement(
+                  household = household,
+                  selectedHousehold = it,
+                  onHouseholdSelected = { household ->
+                    if (household != selectedHousehold) {
+                      householdViewModel.selectHousehold(household)
+                    }
+                    scope.launch { drawerState.close() }
+                  })
+            }
+          }
+          HorizontalDivider(modifier = Modifier.padding(horizontal = 16.dp))
+          Row(
+              modifier = Modifier.fillMaxWidth().padding(16.dp),
+              verticalAlignment = Alignment.CenterVertically,
+              horizontalArrangement = Arrangement.Center) {
+                IconButton(onClick = { showDialog = true }) {
+                  Icon(
+                      imageVector = Icons.Default.Add,
+                      contentDescription = "Add Household Icon",
+                      modifier = Modifier.testTag("addHouseholdIcon"))
+                }
+
+                IconButton(onClick = { showEdit = true }) {
+                  Icon(
+                      imageVector = Icons.Outlined.Edit,
+                      contentDescription = "Edit Household Icon",
+                      modifier = Modifier.testTag("editHouseholdIcon"))
+                }
+              }
+        }
+      },
+  ) {
```

Comment:
Except the `filters`, all this code is a duplication from `Recipes`. For maintainability reason, it might be a good idea to extract it to its own composable. If for some reason a modification must be made it would prevent to forgot to do it to the other place and would maintain consistency between both views.